aws/solutions/TaggingRootVolumesInEC2/Tagging_Root_volume.yaml

```diff
@@ -0,0 +1,149 @@
+   Description: EC2 AZ.
+   Type: AWS::EC2::AvailabilityZone::Name
+   ConstraintDescription: "Must be the name of an availabity zone."
+  WindowsAMIID:
+    Description: The Windows 2016 AMIID, the default value is for the eu-west-1 region
+    Type: String
+    Default: ami-0d138b26f46625e2f
+  LinuxAMIID:
+    Description: The Amazon Linux 2 AMIID, the default value is for the eu-west-1 region
+    Type: String
+    Default: ami-0fad7378adf284ce0
+
+Resources:
+  #Tagging the root volume for the EC2 instances with the Windows OS
+  #The commented out powershell commands are to utilize if the AMI has the AWS CLI installed and you prefer that approach.
+  WindowsInstance:
+    Type: 'AWS::EC2::Instance'
+    Properties:
+      ImageId: !Ref WindowsAMIID
+      InstanceType: !Ref InstanceType
+      AvailabilityZone: !Ref InstanceAZ
+      IamInstanceProfile: !Ref InstanceProfile
+      KeyName: !Ref KeyName
+      UserData: 
+        Fn::Base64: !Sub |
+          <powershell>
+            $AWS_AVAIL_ZONE=(curl http://169.254.169.254/latest/meta-data/placement/availability-zone).Content
+            $AWS_REGION=$AWS_AVAIL_ZONE.Substring(0,$AWS_AVAIL_ZONE.length-1)
+            $AWS_INSTANCE_ID=(curl http://169.254.169.254/latest/meta-data/instance-id).Content
+            $ROOT_VOLUME_IDS=((Get-EC2Instance -Region $AWS_REGION -InstanceId $AWS_INSTANCE_ID).Instances.BlockDeviceMappings | where-object DeviceName -match '/dev/sda1').Ebs.VolumeId
+            $tag = New-Object Amazon.EC2.Model.Tag
+            $tag.key = "MyRootTag"
+            $tag.value = "MyRootVolumesValue"
+            New-EC2Tag -Resource $ROOT_VOLUME_IDS -Region $AWS_REGION -Tag $tag
+          </powershell>
+        #<powershell>
+        #$AWS_AVAIL_ZONE=(curl http://169.254.169.254/latest/meta-data/placement/availability-zone).Content
+        #$AWS_REGION=$AWS_AVAIL_ZONE.Substring(0,$AWS_AVAIL_ZONE.length-1)
+        #$AWS_INSTANCE_ID=(curl http://169.254.169.254/latest/meta-data/instance-id).Content
+        #$ROOT_VOLUME_IDS =(aws ec2 describe-instances --region eu-west-1 --instance-id $AWS_INSTANCE_ID --query Reservations[0].Instances[0].BlockDeviceMappings[0].Ebs.VolumeId)
+        #aws ec2 create-tags --resources $ROOT_VOLUME_IDS --region $AWS_REGION --tags Key=Name,Value=newAMI
+        #</powershell>
+      Tags:
+        - Key: Name
+          Value: !Ref 'AWS::StackName'
+      BlockDeviceMappings:
+        - DeviceName: /dev/sdm
+          Ebs:
+            VolumeType: io1
+            Iops: '200'
+            DeleteOnTermination: 'true'
+            VolumeSize: '10'
+
+  #Tagging the root volume for EC2 instances with an Unix OS with the AWS CLI installed.
+  LinuxInstance:
+    Type: 'AWS::EC2::Instance'
+    Properties:
+      ImageId: !Ref LinuxAMIID
+      InstanceType: !Ref InstanceType
+      AvailabilityZone: !Ref InstanceAZ
+      IamInstanceProfile: !Ref InstanceProfile
+      KeyName: !Ref KeyName
+      UserData: 
+        Fn::Base64: !Sub |

```

Comment:
Same as above. You have a !Sub for your user-data, but you're not doing parameter injection at the CloudFormation level.